packages/hint-create-element-svg/src/hint.ts

```diff
@@ -70,26 +68,28 @@ export default class CreateElementSvgHint implements IHint {
             const results = linter.verify(sourceCode, { rules: { 'svg-create': 'error' } });
 
             for (const result of results) {
-                let position = scriptElement.getLocation() || null;
+                const element = scriptData.element;
+
+                if (element === null) {
+                    return;
+                }
+
+                let position = element.getLocation() || null;
                 let line = 0;
 
                 /*
                  * Eslint returns location data starting at <script>
                  * This offsets the line number so that it is relative
                  * to the full source code and not just the <script> tag
                  */
-                if (scriptElement && scriptData.resource === 'Internal javascript') {
-                    position = await findProblemLocation(scriptElement, { column: 0, line: 0 }, result.source === null ? '' : result.source);
+                if (scriptData.resource === 'Internal javascript') {
+                    position = await findProblemLocation(element, { column: 0, line: 0 }, result.source === null ? '' : result.source);
                     line = position !== null ? position.line : 0;
```

Comment:
Does this work if we have `<script>document.createElement("circle")</script>` all on one line? We may need to adjust `column` in addition to `line`. This probably also means we need to account for the size of the opening script tag, e.g. in `<script type="text/javascript">document.createElement("circle")</script>`.

Reply:
Looks like `parse5` includes [location information for the end of the start tag](https://github.com/inikulin/parse5/blob/master/packages/parse5/docs/source-code-location/start-tag-location.md) of an element. We should be able to extend our internal `HTMLElement` type to expose this once #1943 is merged.